src/public/pages/examples/querystring-params.js

```diff
@@ -0,0 +1,63 @@
+import K, * as U from "karet.util"
+import * as L    from "partial.lenses"
+import React     from "karet"
+
+import Link from "../../components/link"
+
+import { TextInput }       from "../../components/text-input"
+import { PrettyStringify } from "../../components/pretty-stringify"
+
+const addQuerystringParams = params =>
+  params.modify( U.append( [ '', '' ]  ) )
+
+const QuerystringParam = ( { param } ) =>
+  <tr>
+    <td><TextInput value={ U.view( [ 0 ], param ) }/></td>
+    <td><TextInput value={ U.view( [ 1 ], param ) }/></td>
+    <td><button onClick={ () => param.remove() }>Remove</button></td>
+  </tr>
+
+const makeQuerystring =
+  U.ifElse( U.isNil
+          , _ => ''
+          , U.pipe( U.map( U.pipe( U.map( encodeURIComponent )
+                                 , U.join( '=' )
+                                 )
+                         )
+                  , U.join( '&' )
+                  , U.concat( '?' )
+                  )
+          )
+
+const paramsPairs = params =>
+  U.view( L.iso( U.toPairs, U.fromPairs ), params )
+
+const newPathString = ( path, params ) =>
+  U.string`${ path }${ makeQuerystring( params ) }`
+
+export const QuerystringParams = ( { params, path, copy = U.atom( [] ) } ) =>
+  <div>
+    <table>
+      { U.set( copy, paramsPairs( params ) ) }
+      <thead>
+        <tr>
+          <td>Key</td>
+          <td>Value</td>
+          <td>
+            <button onClick={ () => addQuerystringParams( copy ) }>Add</button>
+          </td>
+        </tr>
+      </thead>
+      <tbody>
+        { U.mapElems( ( param, i ) =>
+                        <QuerystringParam key={ i } param={ param }/>
+                    )( copy )
+        }
+      </tbody>
+    </table>
+    Navigate to:&nbsp;
+    <Link href={ newPathString( path, copy ) }>
+      { newPathString( path, copy ) }
```

Comment:
This is obvious, of course, but here the property `newPathString( path, copy )` is computed twice.  This gives you two independent copies of the resulting observable and both of them compute the path string independently after each change.  It doesn't really matter in this case, but it can be better to bind such computations to a variable and use the variable twice to avoid duplicated computation.

BTW, Karet Util has a `scope` function, which can be convenient:

```js
{U.scope((href = newPathString(path, copy)) =>
   <Link href={href}>{href}</Link> )}
```

`U.scope(fn)` is equivalent to `fn()`.

Or you could have a link component that just displays the `href`.